private/byte.rkt

```diff
@@ -109,3 +116,279 @@
+           (bitwise-and (byte-ref left i)
+                        (byte-ref right i))))) 
+
+(module+ test
+  (test-case
+    "byte-and"
+    (for ([x (in-range 256)])
+      (check-equal? (byte-ref (byte-and x (byte 0 1 1 1 1 1 1 1)) 0) 0)
+      (check-equal? (byte-ref (byte-and (byte 0 1 1 1 1 1 1 1) x) 0) 0)
+
+      (check-equal? (byte-ref (byte-and x (byte 1 0 1 1 1 1 1 1)) 1) 0)
+      (check-equal? (byte-ref (byte-and (byte 1 0 1 1 1 1 1 1) x) 1) 0)
+
+      (check-equal? (byte-ref (byte-and x (byte 1 1 0 1 1 1 1 1)) 2) 0)
+      (check-equal? (byte-ref (byte-and (byte 1 1 0 1 1 1 1 1) x) 2) 0)
+
+      (check-equal? (byte-ref (byte-and x (byte 1 1 1 0 1 1 1 1)) 3) 0)
+      (check-equal? (byte-ref (byte-and (byte 1 1 1 0 1 1 1 1) x) 3) 0)
+
+      (check-equal? (byte-ref (byte-and x (byte 1 1 1 1 0 1 1 1)) 4) 0)
+      (check-equal? (byte-ref (byte-and (byte 1 1 1 1 0 1 1 1) x) 4) 0)
+
+      (check-equal? (byte-ref (byte-and x (byte 1 1 1 1 1 0 1 1)) 5) 0)
+      (check-equal? (byte-ref (byte-and (byte 1 1 1 1 1 0 1 1) x) 5) 0)
+      
+      (check-equal? (byte-ref (byte-and x (byte 1 1 1 1 1 1 0 1)) 6) 0)
+      (check-equal? (byte-ref (byte-and (byte 1 1 1 1 1 1 0 1) x) 6) 0)
+
+      (check-equal? (byte-ref (byte-and x (byte 1 1 1 1 1 1 1 0)) 7) 0)
+      (check-equal? (byte-ref (byte-and (byte 1 1 1 1 1 1 1 0) x) 7) 0))))
+
+(define (byte-or left right)
+  (apply byte
+         (for/list ([i (in-range 8)])
+           (bitwise-ior (byte-ref left i)
+                       (byte-ref right i))))) 
+
+(module+ test
+  (test-case
+    "byte-or"
+    (for ([x (in-range 256)])
+      (check-equal? (byte-ref (byte-or x (byte 1 0 0 0 0 0 0 0)) 0) 1)
+      (check-equal? (byte-ref (byte-or (byte 1 0 0 0 0 0 0 0) x) 0) 1)
+
+      (check-equal? (byte-ref (byte-or x (byte 0 1 0 0 0 0 0 0)) 1) 1)
+      (check-equal? (byte-ref (byte-or (byte 0 1 0 0 0 0 0 0) x) 1) 1)
+
+      (check-equal? (byte-ref (byte-or x (byte 0 0 1 0 0 0 0 0)) 2) 1)
+      (check-equal? (byte-ref (byte-or (byte 0 0 1 0 0 0 0 0) x) 2) 1)
+
+      (check-equal? (byte-ref (byte-or x (byte 0 0 0 1 0 0 0 0)) 3) 1)
+      (check-equal? (byte-ref (byte-or (byte 0 0 0 1 0 0 0 0) x) 3) 1)
+
+      (check-equal? (byte-ref (byte-or x (byte 0 0 0 0 1 0 0 0)) 4) 1)
+      (check-equal? (byte-ref (byte-or (byte 0 0 0 0 1 0 0 0) x) 4) 1)
+
+      (check-equal? (byte-ref (byte-or x (byte 0 0 0 0 0 1 0 0)) 5) 1)
+      (check-equal? (byte-ref (byte-or (byte 0 0 0 0 0 1 0 0) x) 5) 1)
+      
+      (check-equal? (byte-ref (byte-or x (byte 0 0 0 0 0 0 1 0)) 6) 1)
+      (check-equal? (byte-ref (byte-or (byte 0 0 0 0 0 0 1 0) x) 6) 1)
+
+      (check-equal? (byte-ref (byte-or x (byte 0 0 0 0 0 0 0 1)) 7) 1)
+      (check-equal? (byte-ref (byte-or (byte 0 0 0 0 0 0 0 1) x) 7) 1))))
+
+(define (byte-not b)
+  (apply byte
+         (for/list ([i (in-range 8)])
+           (bitwise-xor (byte-ref b i) 1)))) 
+
+(module+ test
+  (test-case
+    "byte-not"
+    (for ([x (in-range 256)])
```

Comment:
Another test we could add for `byte-not`: a byte and its negation should always add up to 255.

```
(for ([x (in-range 256)])
  (check-equal? (+ x (byte-not x)) 255))
```